tests/testutil/service.go

```diff
@@ -135,6 +135,43 @@ func (ts *TestService) Attach() []string {
 	return results
 }
 
+// BuildAndCreate builds a local image and runs a service using it
+func (ts *TestService) BuildAndCreate(t *testing.T, imageName string, imageVersion string, args ...string) {
+	err := ts.BuildImage(imageName, imageVersion, args...)
+	if err != nil {
+		ts.T.Fatal(err)
+	}
+	ts.Create(t, fmt.Sprintf("localhost:5442/%s/%s:%s", TestingNamespace, imageName, imageVersion))
+}
+
+// BuildAndStage builds a local image and stages it onto another running service
+func (ts *TestService) BuildAndStage(t *testing.T, imageName string, imageVersion string, args ...string) TestService {
+	err := ts.BuildImage(imageName, imageVersion, args...)
+	if err != nil {
+		ts.T.Fatal(err)
+	}
+	return ts.Stage(fmt.Sprintf("localhost:5442/%s/%s:%s", TestingNamespace, imageName, imageVersion), imageVersion)
+}
+
+// BuildAndCreate builds a local image and runs a service using it
+func (ts *TestService) BuildImage(imageName string, imageVersion string, args ...string) error {
+	pwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("Could not retrieve working directory.  %v", err.Error())
+	}
+	if strings.Contains(pwd, "tests/integration") {
```

Comment:
hmmm does pwd ever change?

Reply:
Not in Drone 😬 

I found that for some reason, `pwd` was always `tests/integration` in Drone or running from the Dockerfile. The other case accounts for running locally which we tend to do from rio root directory.